### PR TITLE
Fix: support Qualified Wildcard in count aggregate function

### LINKED ISSUE #11323: Support Qualified Wildcard in Count
### Is your feature request related to a problem or challenge?

In working on https://github.com/apache/datafusion/pull/11229 I noticed that `SELECT COUNT(t1.*) FROM t1` doesn't work and throws a not implemented error.

```
Error: NotImplemented("Unsupported qualified wildcard argument: Unnamed(QualifiedWildcard(ObjectName([Ident { value: \"t1\", quote_style: None }])))")
```

If it is to be implemented it may make sense to wait until after https://github.com/apache/datafusion/pull/11229 is merged.

### Describe the solution you'd like

Fill out this match arm: https://github.com/apache/datafusion/blob/229c1398d6bcd4267c8f7030a4fcf7d17c096b74/datafusion/sql/src/expr/function.rs#L459

### Describe alternatives you've considered

_No response_

### Additional context

This query is supported by postgres and duckdb as a cursory check, so might be nice to support it, though it's not used super often.

### PR DESCRIPTION
**Pull request by HuSen8891**
## Which issue does this PR close?

Closes #11323

### CODE REVIEW

File: datafusion/sqllogictest/test_files/aggregate.slt [hunk 1]
```diff
@@ -1124,6 +1124,11 @@ SELECT COUNT(*) FROM aggregate_test_100
 ----
 100
 
+query I
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/aggregate.slt:1127]: Can you please also add a negative test (aka a test that shows an unknown table)?

Perhaps something like

```sql
query error
SELECT COUNT(foo.*) FROM aggregate_test_100
```

**Review by alamb — COMMENTED** [review]
Thank you @HuSen8891 for this contribution. I started the CI and left a suggestion for one more comment. Otherwise I think this PR is looking ready to go

File: datafusion/sqllogictest/test_files/aggregate.slt (hunk 1 above)
Comment (HuSen8891) [reply datafusion/sqllogictest/test_files/aggregate.slt:1127]: I'll add this test case late, thanks!

**Review by alamb — APPROVED** [review]
Thank you @HuSen8891  -- this looks great to me

cc @jonahgao  in case you would also like to review

**Review by jonahgao — APPROVED** [review]
LGTM, thanks @HuSen8891

**Comment by alamb** [comment]
🚀

**Pull request merged**